### DISCUSSION TITLE
Forge vs Automatic - April 2024

### DISCUSSION THREAD

**Discussion by thiagojramos** [lllyasviel/stable-diffusion-webui-forge#681] [General]
Currently, after the WebUI of automatic has evolved to version 1.9 and such, and with the lack of Forge updates for a considerable time, as well as the uncertain future of the project, is it still worth using Forge?

I always have difficulty perceiving when these programs have different performance when it comes to a matter of seconds. But lately, I have been finding the Automatic1111 WebUI to be almost the same speed in generating images (except when using ControlNet, among others). That said, where I feel the Forge is more efficient is in RAM usage and model switching.

**Comment by strawberrymelonpanda** [comment]
I certainly want to see Forge updated again, but whether its worth it depends on the person.  

Do you avoid OOM errors at high res because of Forge's smarter offloading, etc?  Is the speedup for your particular card significant?  Do you use Forge-only (vs Auto) tools like Layer Diffuse or IP Adapter masking?  If so, it's probably worth it.  

On the one hand, I'd also argue many of the 1.8 -> 1.9 changes aren't terribly significant for most people compared to the Forge changes.  Some people have even rolled back.  On the other hand, ControlNet for Auto has continued to improve, while Forge's built-in one can not until updates resume.

Then there's the question of how long until SD3 will work with Forge, once it's out.

Personally I've been trying Comfy again.  I _don't_ prefer it over Auto/Forge, but it has Forge's speed with more updates and newer features. If nothing else, it scratches the itch for fooling around with new things while Forge still works reliably.

**Comment by ostap667inbox** [comment]
For me personally, using Forge is necessary at least because of these two things:
- The availability of the 'Never OOM' tab thanks to which I have the ability to generate large resolution images via hires.fix without additional scripts.
- The availability of the Forge Couple plugin, which is still being developed and which can be used to perfectly control the zoning of objects in the scene.

**Reply by thiagojramos** [reply]
I wasn't aware of the 'Forge Couple' extension (I'll check it out). The 'Never OOM' is really essential.

**Reply by ostap667inbox** [reply]
In A1111, the [Regional Prompter](https://github.com/hako-mikan/sd-webui-regional-prompter) plug-in can perform these same functions. However, it is incredibly hideous from a usability standpoint. That's not a figure of speech. If I'm asked to give an example of an interface that is as unfriendly to the user as possible, this plugin will be the first thing that comes to my mind. It's horrible.
This is the one glaring case that proves that a programmer should program and a UI/UX designer should create an interface. And they should not substitute each other's functions.

**Comment by wisdombox44** [comment]
I believe developers have plans to upgrade after SD3 public release. Developer Illyasviel has many active repositories, including control.net and Foocus web UI. I think we must be patient. In the meantime, I play with ComfyUI for new things, even though a node-based workflow is not my personal choice. I am happy with Foocus and Forge web UI. I would like to see some nice features from Foocus to be implemented into Forge web UI, like ready styles, upscaling, inpainting (which is very efficient and handy), face swap, and style transfer - making it very easy. Overall, it has the best web UI UX design. The only downside to Foocus is that it doesn't support extensions and lacks some control.net features. In summary, thank you to all the developers; we are getting everything for free.

**Comment by Diodonea** [comment]
Hello! Even though there are plans for Forge, I believe that, at the very least, the Forge developers should issue some statement so that we know Forge hasn't been abandoned.

**Comment by LadyFlames** [comment]
A1111 has its own problems to face like i just tried using it but all i get is an Error:Bad sampler 
and same with SDForge at the moment some of the integrated extensions Broke SDForge pretty much entirely and they havent abandoned Forge the Collaborators of Forge made a unified decision to Switch Forge to Dev2 Branch and they recomended that all users Switch Their Forge version to Dev2 just keep in mind if you do Switch dont use Control Net or rather any of the integrated Extension of Forge  and  well its more like the multi input aspect of Control net Broke SDForge

**Comment by NergiZed** [comment]
Auto1111's stability problems is almost debilitating, and it's much more difficult to do fresh re-installs. While Forge might blow up, usually at least parts of it still works. Every time I am forced to go back to Auto1111, I usually walk out of it with errors. This is on top of Auto1111 being noticeably slower.

Forge can use around 80% of Auto1111's extensions, and has a set of unique exclusive extensions like forge ic-light, layer diffuse, forge-couple (better region prompt), forge-animatedDiff (this has some extra features compared to the normal webui versoin)

Currently, some integrated extensions do not work in Forge. I'm pretty sure Style-Align is bugged, at least when I try to use it with ControlNet. For me, the most troublesome thing is that some newer ControlNet features are not there, and some batch functions in the Integrated ControlNet are bugged.

I hope these gradually get resolved, but I don't have the technical know-how to comfortably switch to dev2. I'm fine with waiting / just investigating other things in the meantime. Now that lllyasviel has released IC-Light, perhaps he'll return to this project after taking a breather.

**Comment by freecoderwaifu** [comment]
Never really encountered long lasting breaking issues or errors in A1111, only at times of big update bursts which is understandable. I don't doubt they don't happen to some, but it makes me question more how certain environments and workflows are set up. 

Forge is still faster and has support for some exclusive extensions. However, A1111 already had a decent amount of commits like re-adding directory view to extra networks (on top of a big upcoming performance and UI improvement for extra networks), split samplers from schedulers, improved cache, and some security updates, that are enough for me to stay on it, at least until Forge can sync them in.

**Reply by OleJ1964** [reply]
Forge is really great, but more and more combability problems are stacking up so Auto 1111 is again my prefered, but boy do i miss the memory handling from Forge. I have 12GB VRam and Forge runs as smooth as butter, Auto 1111 not so much :-(  Please Auto1111 implement the memory handling from Forge....

**Comment by koopa990** [comment]
I just switched to Forge last month, and I heard Auto 1111 received a latest patch to the system. Maybe it is best for me to stay where I am now.

**Comment by LadyFlames** [comment]
some Forge users are experiencing Error:None Type iterable when using any integrated&External extensions and any Script with Forge but the issues goes away when Extensions and Scripts are not being used no one knows what causes that error other than that it works fine minus the occasional Hires Fix Slowdowns like this morning took 7hours to generate 1 image at 512x512

**Reply by rbrtwtrs** [reply]
Those None Type errors are due to having something missing like an image for a Controlnet or ReActor model or a value that is out of scope. Unfortunately it is often hard and time consuming to figure out what is causing it. The error log isn't very helpful in this regard. There are so many things that can cause it and misconfiguring an extension can have you tearing your hair out. Deselect extensions and start with basics. A verbose error log mode would help.
Another clue about cause is where the error happens. Is it right away or at a post processor stage.

**Reply by LadyFlames** [reply]
[sysinfo-2024-06-03-15-57.json](https://github.com/user-attachments/files/15536570/sysinfo-2024-06-03-15-57.json)
i have a feeling its control net or the modules themselves thats the problem dont know how to properly read the syslog

**Reply by rbrtwtrs** [reply]
You can try deselecting extensions one at a time and see if that helps. I had a typo in my Config Preset Extensions config-img2img.json  file that caused the error every time I tried to run an img to img. So easy to make a minor error and screw the whole setup. Mismatching controlnet models or aspect ratios will also cause the same errors.
Often times extensions have a Forge specific setup or are not Forge compatible and just generic installing from the extensions tab can bork things. Have to read the fine print.

**Comment by koopa990** [comment]
Unless auto--or even ComfyUI--has fixed my end issues where they consist of grey genned images, and progress hang up on genning and upscaling images.

**Comment by redrum-llik** [comment]
I have an Intel Arc 16 GB VRAM card, and Automatic/SD.Next will often OOM on just 3-4 items batch with SDXL at default resolution. Forge, on the other hand, can handle anything given enough time. If this level of memory management could be achieved in base A1111, I would probably consider moving over (provided that Forge Couple has an alternate implementation for A1111, of course). 

That being said, is anyone aware of the extensions that handle the model offloading/other functionality provided by NeverOOM?
